### DISCUSSION TITLE
Paste from clipboard into machine

### DISCUSSION THREAD

**Discussion by randompersononinternet69** [86Box/86Box#4946] [Ideas]
i think there should be a paste from clipboard function like DOSBox-X has. DOSBox-X just types out the clipboard contents as fast as it can to create a CTRL+V like experience.

**Comment by Pellzi** [comment]
My wish to this Paste from Host Clipboard into Machine:

The Keyboard Key Combination to do so, should be: RIGHT SHIFT + LEFT CTRL + V.
And optionaly in the Action Tab above the Primary Output Windows of 86Box: Insert Host Clipboard (RIGHT SHIFT + LEFT CTRL +V)

If it is only CTRL + V, i see problems, when this Key Combination is pressed during work inside the machine.

Probably it possible to have also this List entry in the Action Tab:
Copy Clipboard of Machine to Host. (RIGHT SHIFT + LEFT CTRL + C)

**Reply by randompersononinternet69** [reply]
Copy clipboard from emulated PC to host PC sounds nice as well, but it would require special drivers to communicate between the host and emulated PC.

**Comment by Pellzi** [comment]
Are you able, to write such a Driver? Or perhaps you know a person, who can do it?
Probably, it´s not necessary, to special configure such a new Driver only for 86Box, instead generaly for Emulators?
I know many Emulators, for very different OS. But practicaly, no one of them can Copy clipboard from emulated PC to host PC.
Thanks for the answer, now i know why.

**Reply by seal331** [reply]
AFAIK there's no specified widely accepted standard in emulators on something like "guest-to-host clipboard sharing interface", so no, making a driver which will work on emulators generally is just technically impossible (not to mention it requires work on the emulator side as well). There's also the fact that you would probably have to write that driver for every OS under the sun - not everything is as dedicated to backwards compatibility as Windows.

**Reply by randompersononinternet69** [reply]
you can't user drivers for clipboard contents to copy from guest -> host, and host -> guest. see seal331's reply

**Comment by Emayeah** [comment]
Hyper-V does have this feature too

**Reply by randompersononinternet69** [reply]
also VMware has this. these are both virtual machine software though, not emulators.
